provider/azure/environ.go

```diff
@@ -1074,7 +1074,56 @@ func (env *azureEnviron) instances(
 
 // AdoptResources is part of the Environ interface.
 func (env *azureEnviron) AdoptResources(controllerUUID string, fromVersion version.Number) error {
-	return errors.NotImplementedf("AdoptResources")
+	groupClient := resources.GroupsClient{env.resources}
+	resourceClient := resources.Client{env.resources}
+	var failed []string
+
+	resources, err := groupClient.ListResources(env.resourceGroup, "", nil)
+	if err != nil {
+		return errors.Annotate(err, "listing resources")
+	}
+	for resources.Value != nil {
+		for _, resource := range *resources.Value {
+
+			providerNamespace, parentPath, err := splitResourceType(*resource.Type)
```

Comment:
Ah, actually I don't think this bit is right.

Microsoft.Compute is the namespace, availabilitySets is the resource type; but it is not a *parent* resource type, as availability sets don't have a parent.

If there were other bits in between Microsoft.Compute and availabilitySets, the other bits would constitute the parent path.